### PR TITLE
[EDLY-1932] Oauth2 Provider confugration of same backend for multiple sites

### PR DESCRIPTION
**Pull request by ziafazal**
Added abilit to create Oauth2 Provider confugration of same backend for multiple sites.

### CODE REVIEW

File: common/djangoapps/third_party_auth/tests/test_provider.py [hunk 1]
```diff
@@ -146,6 +148,16 @@ def test_provider_disabled_for_mismatching_site(self):
         prov = self.configure_google_provider(visible=True, enabled=True, site=site_b)
         self.assertEqual(prov.enabled_for_current_site, False)
 
+    @ddt.data(SITE_DOMAIN_A, SITE_DOMAIN_B)
+    def test_single_provider_for_multiple_sites(self, site_domain):
+        """
+        Verify that enabled_for_current_site returns True when the provider is configured for a multiple sites.
```
Comment (zubair-arbi) [inline common/djangoapps/third_party_auth/tests/test_provider.py:154]:
```python
@ddt.data(SITE_DOMAIN_A, SITE_DOMAIN_B)
def test_single_provider_for_multiple_sites(self, site_domain):
    """
    Verify that "enabled_for_current_site" returns True when the auth provider is configured for multiple sites.
    """
    site, __ = Site.objects.get_or_create(domain=site_domain, name=site_domain)
    auth_provider = self.configure_google_provider(visible=True, enabled=True, site=site)
    with patch('django.contrib.sites.models.SiteManager.get_current', return_value=site):
        assert auth_provider.enabled_for_current_site
```

**Review by zubair-arbi — APPROVED** [review]
Just one suggestion related to test otherwise changes looks good. 👍 
Thanks for fixing this third party auth issue.

File: lms/envs/devstack.py [hunk 2]
```diff
@@ -260,6 +260,12 @@ def should_show_debug_toolbar(request):
     ),
 })
 
+#####################################################################
+
+# django-session-cookie middleware
+DCS_SESSION_COOKIE_SAMESITE = 'Lax'
```
Comment (rehan99000) [inline lms/envs/devstack.py:266]: These changes are unrelated to this PR

File: openedx/features/edly/utils.py [hunk 3]
```diff
@@ -147,17 +147,18 @@ def create_user_link_with_edly_sub_organization(request, user):
...
+    edly_user_profile = None
     try:
         edly_sub_org = request.site.edly_sub_org_for_lms
     except EdlySubOrganization.DoesNotExist:
         edly_sub_org = request.site.edly_sub_org_for_studio
-    edly_user_profile, __ = EdlyUserProfile.objects.get_or_create(user=user)
-    edly_user_profile.edly_sub_organizations.add(edly_sub_org)
-    edly_user_profile.save()
-
+    if edly_sub_org:
```
Comment (rehan99000) [inline openedx/features/edly/utils.py:158]: Similarly, I think these changes are about something else and should be done in a separate PR.

**Review by rehan99000 — CHANGES_REQUESTED** [review]
2 unrelated changes otherwise looks good.

File: common/djangoapps/third_party_auth/strategy.py [hunk 4]
```diff
@@ -29,7 +32,8 @@ def setting(self, name, default=None, backend=None):
             setting 'name' is configured via LTIProviderConfig.
         """
         if isinstance(backend, OAuthAuth):
-            provider_config = OAuth2ProviderConfig.current(backend.name)
+            site = Site.objects.get_current(get_current_request())
```
Comment (iam-mhaseeb) [inline common/djangoapps/third_party_auth/strategy.py:35]: Fancy 👍

File: openedx/features/edly/utils.py (hunk 3 above)
Comment (iam-mhaseeb) [inline openedx/features/edly/utils.py:158]: What if there is no site for the studio too? Would it break something?

**Review by iam-mhaseeb — APPROVED** [review]
Thanks for the fix! 🙏 
Added 1 question. Otherwise changes LGTM 👍

**Comment by ziafazal** [comment]
@zubair-arbi addressed your feedback.
@rehan99000 removed unrelated changes.

**Review by rehan99000 — APPROVED** [review]

File: common/djangoapps/third_party_auth/tests/test_provider.py (hunk 1 above)
Comment (zubair-arbi) [reply common/djangoapps/third_party_auth/tests/test_provider.py:154]: @ziafazal Please replace `test_single_provider_for_multiple_sites` with the above code.
